### PR TITLE
xorg-proto: Add compatibility to v2

### PR DESCRIPTION
**Pull request by RubenRBS**
Specify library name and version:  **xorg-proto/all**

Needed by imake in #15880 for v2, needs xorg-macros #16008

### CODE REVIEW

File: recipes/xorg-proto/all/conanfile.py [hunk 1]
```diff
@@ -40,7 +39,7 @@ def build_requirements(self):
         self.build_requires("automake/1.16.3")
         self.build_requires("xorg-macros/1.19.3")
         self.build_requires("pkgconf/1.7.4")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+        if self._settings_build.os == "Windows" and not os.environ["CONAN_BASH_PATH"]:
```
Comment (prince-chrismc) [inline recipes/xorg-proto/all/conanfile.py:42]: 🤔 should be a conf check now AFAIK

File: recipes/xorg-proto/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake
+
 import os
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake"
+    generators = "CMakeDeps", "AutotoolsDeps"
```
Comment (prince-chrismc) [inline recipes/xorg-proto/all/test_package/conanfile.py:10]: Autotools Deps but no CMakeToolchian?

File: recipes/xorg-proto/all/conanfile.py [hunk 3]
```diff
@@ -53,45 +52,38 @@ def package_id(self):
         del self.info.settings.build_type
```
Comment (prince-chrismc) [inline recipes/xorg-proto/all/conanfile.py:52]: Should probably be package type application?

File: recipes/xorg-proto/all/conanfile.py [hunk 4]
```diff
@@ -53,45 +52,38 @@ def package_id(self):
...
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        env = tc.environment()
+        if is_msvc(self):
+            compile_wrapper = unix_path(self, self.conf.get("user.automake:compile-wrapper"))
+            env.define("CC", f"{compile_wrapper} cl -nologo")
+        tc.generate(env)
+
+        autotools = Autotools(self)
+        autotools.configure(build_script_folder=self.source_folder)
```
Comment (prince-chrismc) [inline recipes/xorg-proto/all/conanfile.py:70]: Should be n the build method :)

File: recipes/xorg-proto/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/xorg-proto/all/conanfile.py:42]: https://github.com/conan-io/conan-center-index/blob/044e188e95c82b9ff0e9bb71fdc9e1c9618b15c7/recipes/gumbo-parser/all/conanfile.py#L55-L58

File: recipes/xorg-proto/all/conanfile.py [hunk 5]
```diff
@@ -1,97 +1,86 @@
...
-        self.build_requires("automake/1.16.3")
-        self.build_requires("xorg-macros/1.19.3")
-        self.build_requires("pkgconf/1.7.4")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        self.tool_requires("automake/1.16.5")
+        self.tool_requires("xorg-macros/1.19.3")
+        self.tool_requires("pkgconf/1.9.3")
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", check_type=str):
+            self.tool_requires("msys2/cci.latest")
```
Comment (StellaSmith) [inline recipes/xorg-proto/all/conanfile.py:39]:
```suggestion
        if self._settings_build.os == "Windows":
        self.win_bash = True
        if not self.conf.get("tools.microsoft.bash:path", check_type=str):
            self.tool_requires("msys2/cci.latest")
```

**Review by StellaSmith — CHANGES_REQUESTED** [review]

**Review by StellaSmith — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**